### ISSUE TITLE
Missing timestamps and no logfile.csv creation

### ISSUE THREAD

**Issue by xYantix** [kamsec/crypto-bot#1]
Hi there,

I was having this problem on a clean Ubuntu VM so I set up a Pi 4 but I'm seeing the same behaviour. 

Haven't added production API details yet so it's just running via testnet configuration.

With the first run of bot.py I get this error.
```
pi@raspberrypi:~/crypto-bot $ python3 bot.py 
Traceback (most recent call last):
  File "bot.py", line 82, in <module>
    main()
  File "bot.py", line 43, in main
    update_successful = update_all_data(fill_missing=False)
  File "/home/pi/crypto-bot/src/data_update.py", line 89, in update_all_data
    new_df = update_ratio(ratio, filename, request_records=336, fill_missing=fill_missing)
  File "/home/pi/crypto-bot/src/data_update.py", line 77, in update_ratio
    raise Exception(f'Missing {len(missing_ts)} timestamps. Update skipped, bot set inactive.')
Exception: Missing 2 timestamps. Update skipped, bot set inactive.
```
If I run it again I don't get an error and logfile.log is created (Still no .csv though)

```
pi@raspberrypi:~/crypto-bot $ cat logfile.log 
2021-10-04 00:26:20 [INFO] Bot started
2021-10-04 00:26:22 [INFO] [BALANCE] 10000.00 USD, 1.0000000 BTC
2021-10-04 00:26:22 [ERROR] [UPDATE] Missing 2 timestamps. Update skipped, check manually.
2021-10-04 00:26:22 [INFO] [CONFIG] STATUS_ACTIVE set to False
```
When I try to launch app.py it errors out and I think it's because logfile.csv doesn't exist. Do you have any ideas about how I can get this going?

**Comment by kamsec** [comment]
It happens few times per year that Binance API returns no data about prices/volumes for some hours. I have implemented interpolation in case of missing values, but it was disabled by default. I simply didn't want the bot being active if Binance API keeps sending "missing data" for long period of time.

I looked at data received by Binance, and this error was caused by these entries:
![img1](https://user-images.githubusercontent.com/62106309/136055630-7b4b2b8d-89eb-4942-9f68-e2f9ebf29d20.PNG)

There is linear interpolation method implemented, so data that caused the issue, after interpolation look like this:
![img2](https://user-images.githubusercontent.com/62106309/136056035-afc1819d-51ce-44af-85cc-e0e4d6159851.PNG)

I also changed the bot behaviour so now it interpolates missing values by default, and logs this fact in logfile.log. If Binance will return missing data for long period of time, this will be visible in logs and action can be taken. It should be solved now!
